includes/class-widget.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * HM Content TOC Widget
+ */
+
+namespace HM\Content_TOC;
+use WP_Widget;
+
+// Abort - if this file is accessed directly and not via WP
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class Widget extends WP_Widget {
+
+	// Main TOC singleton object
+	protected $toc;
```

Comment:
I don't see why not. As long as you remember to test the widget when you make changes to the class.